src/lib/getSlug.ts

```diff
@@ -0,0 +1,23 @@
+import fs from 'fs';
+import path from 'path';
+
+import grayMatter from 'gray-matter';
+
+import { debug } from './logger';
+
+/**
+ * Returns the slug of the specified Markdown or HTML file
+ *
+ * @param {String} filepath path to the HTML/Markdown file
+ *  (file extension must end in `.html`, `.md`., or `.markdown`)
+ * @returns {String} a string containing the slug of the file
+ */
+export default function getSlug(filepath: string) {
+  debug(`reading file ${filepath}`);
+  const file = fs.readFileSync(filepath, 'utf8');
+  const matter = grayMatter(file);
+  debug(`frontmatter for ${filepath}: ${JSON.stringify(matter)}`);
+
+  // Stripping the subdirectories and markdown extension from the filename and lowercasing to get the default slug.
+  return matter.data.slug || path.basename(filepath).replace(path.extname(filepath), '').toLowerCase();
```

Comment:
This logic is a duplication of this logic:

https://github.com/readmeio/rdme/blob/0292a09eabc9e0beb0e982f6f24bed4c2f757a41/src/lib/pushDoc.ts#L34-L41

What if this function was renamed to `readDoc`[^1] and it returned an object containing the `slug`, the `matter` data, and the `file`? Then you could reuse this function in `pushDoc`!

[^1]: `getDoc` would be confusing since it's reading a local file as opposed to making an API request

Reply:
Great idea! Done.